skimage/color/colorconv.py

```diff
@@ -790,6 +790,8 @@ def rgb2gray(rgb):
     """
 
     if rgb.ndim == 2:
+        warn('Gray scale image conversion is now deprecated.',
```

Comment:
Honest question, do we want to raise an error in the future? 

I guess for a line scanner a 1D image with RGB channels make sense.

Maybe we want to specify how we will be interpreting this in the future?

Reply:
I think this warning messages needs to be fleshed out more.

```
The behavior of rgb2gray will change in scikit-image 0.19. Currently, rgb2gray allows 2D grayscale image to be passed as inputs and leaves them unmodified as outputs. In version 0.19, this will no longer be these images will be treated as 1D images with 3 channels.
```

Reply:
I supposed that the new behavior would be described when implemented, in #4418 :-)

Reply:
I think that an error should be raised if the last dimension is not 3, what ever is the number of dimensions